builtin/providers/aws/resource_aws_network_interface.go

```diff
@@ -0,0 +1,307 @@
+	return &schema.Resource{
+		Create: resourceAwsNetworkInterfaceCreate,
+		Read:   resourceAwsNetworkInterfaceRead,
+		Update: resourceAwsNetworkInterfaceUpdate,
+		Delete: resourceAwsNetworkInterfaceDelete,
+
+		Schema: map[string]*schema.Schema{			
+			
+			"subnet_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,				
+				ForceNew: true,
+			},
+
+			"private_ips": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				ForceNew: true,				
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set: func(v interface{}) int {
+					return hashcode.String(v.(string))
+				},
+			},
+
+			"security_groups": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Computed: true,				
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set: func(v interface{}) int {
+					return hashcode.String(v.(string))
+				},
+			},
+
+			"attachment": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"instance": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"device_index": &schema.Schema{
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+						"attachment_id": &schema.Schema{
+							Type:     schema.TypeString,	
+							Computed: true,						
+						},
+					},
+				},
+				Set: resourceAwsEniAttachmentHash,
+			},
+
+			"tags": tagsSchema(),			
+		},
+	}
+}
+
+func resourceAwsNetworkInterfaceCreate(d *schema.ResourceData, meta interface{}) error {
+	
+	ec2conn := meta.(*AWSClient).awsEC2conn
+
+	request := &ec2.CreateNetworkInterfaceRequest{		
+		Groups: 				expandStringList(d.Get("security_groups").(*schema.Set).List()),
+		SubnetID:				aws.String(d.Get("subnet_id").(string)),
+		PrivateIPAddresses:		convertToPrivateIPAddresses(d.Get("private_ips").(*schema.Set).List()),
+	}
+	
+	log.Printf("[DEBUG] Creating network interface")
+	resp, err := ec2conn.CreateNetworkInterface(request)
+	if err != nil {
+		return fmt.Errorf("Error creating ENI: %s", err)
+	}
+
+	new_interface_id := *resp.NetworkInterface.NetworkInterfaceID
+	d.SetId(new_interface_id)
+	log.Printf("[INFO] ENI ID: %s", d.Id())
+
+	return resourceAwsNetworkInterfaceUpdate(d, meta)	
+}
+
+func resourceAwsNetworkInterfaceRead(d *schema.ResourceData, meta interface{}) error {
+
+	ec2conn := meta.(*AWSClient).awsEC2conn
+	describe_network_interfaces_request := &ec2.DescribeNetworkInterfacesRequest{		
+		NetworkInterfaceIDs:	[]string{d.Id()},
+	}
+	describeResp, err := ec2conn.DescribeNetworkInterfaces(describe_network_interfaces_request)
+
+	if err != nil {
+		if ec2err, ok := err.(aws.APIError); ok && ec2err.Code == "InvalidNetworkInterfaceID.NotFound" {
+			// The ENI is gone now, so just remove it from the state
+			d.SetId("")
+			return nil
+		}
+		
+		return fmt.Errorf("Error retrieving ENI: %s", err)
+	}
+	if len(describeResp.NetworkInterfaces) != 1 {
```

Comment:
Re-reading, we're calling `DescribeNetworkInterfacesRequest` with a specific `id`, so this should return 1 or 0, right? In which case, ignore me :smile: